apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx

```diff
@@ -115,34 +127,307 @@ const PostgrestConfig = () => {
         }
       }) ?? []
 
-  const isGraphqlExtensionEnabled =
-    (extensions ?? []).find((ext) => ext.name === 'pg_graphql')?.installed_version !== null
+  function resetForm(values?: Partial<z.infer<typeof formSchema>>) {
+    const dbSchema =
+      config?.db_schema && config?.db_schema ? config.db_schema.replace(/ /g, '').split(',') : []
+
+    if (values?.enableDataApi && dbSchema.length === 0) {
+      dbSchema.push('public', 'storage')
+    }
+
+    const defaultValues = {
+      dbSchema: values?.dbSchema ?? dbSchema,
+      maxRows: values?.maxRows ?? config?.max_rows,
+      dbExtraSearchPath: values?.dbExtraSearchPath ?? config?.db_extra_search_path,
+      dbPool: values?.dbPool ?? config?.db_pool,
+      enableDataApi: values?.enableDataApi,
+    }
 
-  function resetForm() {
     if (config) {
       form.reset({
-        dbSchema: config.db_schema ? config.db_schema.replace(/ /g, '').split(',') : [],
-        maxRows: config.max_rows,
-        dbExtraSearchPath: config.db_extra_search_path,
-        dbPool: config.db_pool,
+        ...defaultValues,
       })
     }
   }
 
+  const onSubmit = async (values: z.infer<typeof formSchema>) => {
+    if (!projectRef) return console.error('Project ref is required') // is this needed ?
+
+    updatePostgrestConfig({
+      projectRef,
+      dbSchema: values.dbSchema.join(', '),
+      maxRows: values.maxRows,
+      dbExtraSearchPath: values.dbExtraSearchPath,
+      dbPool: values.dbPool ? values.dbPool : null,
+    })
+  }
+
   useEffect(() => {
-    if (schema && config) {
-      resetForm()
+    if (config) {
+      /**
+       * Checks if enableDataApi should be enabled or disabled
+       * based on the db_schema value being empty string
+       */
+      const enableDataApi = config.db_schema ? true : false
+      /**
+       * Reset the form to default values
+       */
+      resetForm({ enableDataApi })
     }
-  }, [schema && config])
+    // ignore dep array warning, as cant have resetForm as a dependency
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [config])
+
+  const isDataApiEnabledInForm = form.getValues('enableDataApi')
 
   return (
     <Form_Shadcn_ {...form}>
       <form id={formId} onSubmit={form.handleSubmit(onSubmit)}>
-        <FormPanel
-          disabled={true}
-          header={<p>Data API Settings</p>}
-          footer={
-            <div className="flex px-8 py-4">
+        <FormPanelContainer>
+          <FormPanelHeader>Data API Settings</FormPanelHeader>
+          <FormPanelContent>
+            {isError ? (
+              <Admonition type="destructive" title="Failed to retrieve API settings" />
```

Comment:
just want to double check - should we use Admonition or AlertError
or update AlertError to use Admonition?

The latter has the cta to contact support and also surfaces the error from the network request to the user - i reckon we use AlertError?

Reply:
Admonition is a wrapper of Alert. same thing

Reply:
i'm referring to the dashboard component <AlertError> though, not the UI component <Alert_Shadcn>

Reply:
ah, actually your right. I'll revert it

Reply:
sorry bit lost, what do you mean AlertError has a cta ?

Reply:
updated to Alert